docs/doc.md

```diff
@@ -2002,6 +2003,59 @@ func ListHandler(s *Service) func(ctx *gin.Context) {
 }
 ```
 
+### Bind Query with custom unmarshalers
+
+Any structure that has custom `UnmarshalJSON` or `UnmarshalText` or `UnmarshalBinary` can be used to parse input as necessary
+
+```go
+package main
+import (
+  "fmt"
+  "net/http"
+  "strings"
+  "github.com/gin-gonic/gin"
+)
+// Booking contains data binded using custom unmarshaler.
+type Payload struct {
+  Email  EmailDetails `form:"email"`
+}
+// this structure has special json unmarshaller, in order to parse email (as an example) as specific structure
+type EmailDetails struct {
+  Name string
+  Host string
+}
+func (o *EmailDetails) UnmarshalJSON(data []byte) error {
+  elems := strings.Split(string(data), "@")
+  if len(elems) != 2 {
+    return fmt.Errorf("cannot parse %q as email", string(data))
+  }
+  o.Name = elems[0]
+  o.Host = elems[1]
+  return nil
+}
+func main() {
+  route := gin.Default()
+  route.GET("/email", getEmail)
+  route.Run(":8085")
+}
+func getEmail(c *gin.Context) {
+  var p Payload
+  if err := c.ShouldBindQuery(&p); err == nil {
+    c.JSON(http.StatusOK, gin.H{"message": "Email information is correct"})
+  } else {
+    c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+  }
+}
+```
+
+```console
+$ curl "localhost:8085/email?email=test@example.org"
```

Comment:
How come request query parameters are read into a json unmarshaller? What happens when json is posted in curl body?